src/main/java/com/myhexin/cragent/coderabbittest/AlarmChecker.java

```diff
@@ -0,0 +1,28 @@
+package com.myhexin.cragent.coderabbittest;
+
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class AlarmChecker {
+    
+    private static final ConcurrentHashMap<AlarmEvent, Long> EVENT_MAP = new ConcurrentHashMap<>();
+
+    public boolean needAlarm(AlarmModule module, String identity) {
+        AlarmEvent alarmEvent = new AlarmEvent(module, identity);
+        if (module.getExpiredSeconds() == 0) {
+            EVENT_MAP.put(alarmEvent, alarmEvent.createdNanoTime);
+            return true;
+        } else {
+            Long eventNanoTime = EVENT_MAP.compute(alarmEvent, (event, oldCreatedNanoTime) -> {
+                if (isExistedEventExpire(event, oldCreatedNanoTime)) {
```

Comment:
**Issue**: The method isExistedEventExpire is not defined in the provided code snippet. It seems to be missing or not included in the diff. <br> **Fix**: Make sure to include the implementation of the isExistedEventExpire method or handle the case when it is not available. <br> **Code Suggestion**: 
 ```
 private boolean isExistedEventExpire(AlarmEvent event, Long oldCreatedNanoTime) {
     // Implement the logic to check if the event has expired
     // based on the event and oldCreatedNanoTime
     // Return true if the event has expired, false otherwise
 }
 ```